Code/GraphMol/MarvinParse/MarvinDefs.cpp

```diff
@@ -0,0 +1,2978 @@
+    }
+    this->sgroups.clear();
+
+    parent->sgroups.erase(
+        std::find(parent->sgroups.begin(), parent->sgroups.end(), this));
+    delete this;
+  } catch (const std::exception &e) {
+    throw;
+  }
+}
+
+void MarvinMulticenterSgroup::processOneMulticenterSgroup() {
+  MarvinMolBase *actualParent = this->parent;
+  while (actualParent->role() == "MultipleSgroup") {
+    actualParent = actualParent->parent;
+  }
+
+  // delete the bonds to the dummy atom
+  std::vector<MarvinBond *> orphanedBonds;  // list of bonds to delete
+  for (auto orphanedBond : actualParent->bonds) {
+    std::string attachedAtomId = "";
+    if (orphanedBond->atomRefs2[0] == this->center->id ||
+        orphanedBond->atomRefs2[1] == this->center->id) {
+      orphanedBonds.push_back(orphanedBond);
+    }
+  }
+  for (auto orphanedBond : orphanedBonds) {
+    delete orphanedBond;
+    auto orphanedBondIter = std::find(actualParent->bonds.begin(),
+                                      actualParent->bonds.end(), orphanedBond);
+    actualParent->bonds.erase(orphanedBondIter);
+  }
+
+  auto centerIter = std::find(actualParent->atoms.begin(),
+                              actualParent->atoms.end(), this->center);
+  if (centerIter !=
+      actualParent->atoms.end())  // it might already have been deleted by
+                                  // another multicenter group
+  {
+    auto centerPtr = *centerIter;
+    for (auto thisParent = parent;; thisParent = thisParent->parent) {
+      thisParent->atoms.erase(find(
+          thisParent->atoms.begin(), thisParent->atoms.end(),
+          centerPtr));  // get rid of the atoms pointer to the old dummy atom
+      if (thisParent->role() == "MultipleSgroup") {
+        auto thisMultipleParent = (MarvinMultipleSgroup *)thisParent;
+        auto deleteIter =
+            find(thisMultipleParent->parentAtoms.begin(),
+                 thisMultipleParent->parentAtoms.end(), centerPtr);
+        thisMultipleParent->parentAtoms.erase(deleteIter);
+      }
+
+      if (thisParent == actualParent) {
+        break;
+      }
+    }
+    delete centerPtr;  // get rid of the MolAtom
+  }
+
+  // erase the multicenter group from its parent
+
+  for (auto thisParent = parent;; thisParent = thisParent->parent) {
+    thisParent->sgroups.erase(std::find(thisParent->sgroups.begin(),
+                                        thisParent->sgroups.end(), this));
+
+    if (thisParent == actualParent) {
+      break;
+    }
+  }
+
+  delete this;
+}
+
+void MarvinMolBase::prepSgroupsForRDKit() {
+  // this routine recursively fixes the hierarchy of sgroups - some may NOT
+  // actaully belong underneath their parent and can be promnoted to the
+  // grandparent
+
+  // first fix all the children
+
+  std::vector<std::string> sgroupsMolIdsDone;
+  for (bool allDone = false;
+       !allDone;)    // until all at this level are done - but fixing one child
+                     // can add sgroups to this level
+  {
```

Comment:
Doesn't need to be fixed in this PR, but should be kept in mind for future ones,

This - putting the comment before the bracket - is not the documentation style we use. It should be either before the `for` or inside the bracket. 

This also applies to `if` and `else`

Reply:
I changed this one